dewy/common/extract.py

```diff
@@ -55,8 +56,31 @@ def extract_from_pdf(
     return ExtractResult(text=text)
 
 
-async def extract(
-    url: str, *, extract_tables: bool = False, extract_images: bool = False
+async def extract_content(
+    content: bytes, *, extract_tables: bool = False, extract_images: bool = False
+) -> ExtractResult:
+    logger.info("Extracting content from {} bytes", len(content))
+    import filetype
+
+    mime = filetype.guess(content).mime
+    logger.debug("Inferred mime type: {}", mime)
+    match mime:
+        case "application/pdf":
+            return extract_from_pdf(
+                content, extract_tables=extract_tables, extract_images=extract_images
+            )
+        case unrecognized:
+            raise HTTPException(
```

Comment:
Isn't this generally being run async? Seems like this should update the document's error message.

Reply:
Yes, but any exceptions would bubble up to the entry point of the background task, which catches them and updates the document's error message:

https://github.com/DewyKB/dewy/blob/main/dewy/document/router.py#L27

It may mean that this should be a `ValueError` or something, rather than an `HTTPException`.